### PR TITLE
uboot-sunxi: bump to 2023.04

### PR DESCRIPTION
**Pull request by wigyori**
Compile-tested: all boards
Runtime-tested:
 - Cortex-A8: pcDuino
 - Cortex-A7: Bananapro, Bananapi M3, OrangePi 2
 - Cortex-A53:Pine64+

Notes:
 - binman tries to add firmware for the SCP (system control processor), which we don't build, and is optional for the boot process on 64-bit. The SCP firmware (crust) could only be built with a separate toolchain for or1k anyway. Disable this via setting the SCP envvar to /dev/null. For further info, see [1] .
 - Upstreamed patches have been dropped.

[1] https://github.com/u-boot/u-boot/blob/master/board/sunxi/README.sunxi64

Signed-off-by: Zoltan HERPAI <wigyori@uid0.hu>

### CODE REVIEW

**Comment by hauke** [comment]
Does this need swig to build? I think it is ok if we require that now.

**Comment by wigyori** [comment]
> Does this need swig to build? I think it is ok if we require that now.

Yes, it does - there are other u-boot builds which already require it, isn't it?

**Comment by hauke** [comment]
I think mediatek u-boot also needs swig since some time and it is included in the build docker container since multiple years.

**Comment by ynezz** [comment]
> It was hard to check for swig only conditionally in some targets.

FYI https://github.com/openwrt/openwrt/pull/12974/commits/14fa6492cc157cc150a60d0b597e0909b1482f79

File: package/boot/uboot-sunxi/patches/003-add-theobroma-a31-pangolin.patch [hunk 1]
```diff
@@ -371,5 +371,5 @@
 +
```
Comment (hauke) [inline package/boot/uboot-sunxi/patches/003-add-theobroma-a31-pangolin.patch:371]: It looks like `SUNXI_PANGOLIN` is not used at all. I think you can remove the Kconfig file change.
Maybe do it in a separate commit before the u-boot update.

File: package/boot/uboot-sunxi/patches/100-sun6i-alternate-on-UART2.patch [hunk 2]
```diff
@@ -6,11 +6,11 @@ Subject: sun6i: define alternate-function for UART2 on GPG
...
 +++ b/arch/arm/include/asm/arch-sunxi/gpio.h
-@@ -190,6 +190,7 @@ enum sunxi_gpio_number {
+@@ -171,6 +171,7 @@ enum sunxi_gpio_number {
+ #define SUN5I_GPG_SDC1		2
  #define SUN6I_GPG_SDC1		2
  #define SUN8I_GPG_SDC1		2
- #define SUN6I_GPG_TWI3		2
-+#define SUN6I_GPG_UART2         2
++#define SUN6I_GPG_UART2	2
+ #define SUN8I_GPG_UART1		2
```
Comment (hauke) [inline package/boot/uboot-sunxi/patches/100-sun6i-alternate-on-UART2.patch:14]: This patch is only used by the next one `package/boot/uboot-sunxi/patches/101-sun6i-support-console-on-UART2.patch` you can remove this too if the remove `package/boot/uboot-sunxi/patches/101-sun6i-support-console-on-UART2.patch`

File: package/boot/uboot-sunxi/patches/130-fix-mkimage-host-build.patch [hunk 3]
```diff
@@ -0,0 +1,24 @@
+--- a/tools/image-host.c
++++ b/tools/image-host.c
+@@ -1125,6 +1125,7 @@ static int fit_config_add_verification_d
+  * 2) get public key (X509_get_pubkey)
+  * 3) provide der format (d2i_RSAPublicKey)
+  */
++#ifdef CONFIG_TOOLS_LIBCRYPTO
```
Comment (hauke) [inline package/boot/uboot-sunxi/patches/130-fix-mkimage-host-build.patch:7]: Please add a description to the patch.
Are you adding this to work around a compile error you are getting with libressl?
Please post the compile error you get.

File: package/boot/uboot-sunxi/patches/211-no-kwbimage.patch [hunk 4]
```diff
@@ -0,0 +1,10 @@
+--- a/tools/Makefile
++++ b/tools/Makefile
+@@ -113,7 +113,6 @@ dumpimage-mkimage-objs := aisimage.o \
+ 			imximage.o \
+ 			imx8image.o \
+ 			imx8mimage.o \
+-			kwbimage.o \
```
Comment (hauke) [inline package/boot/uboot-sunxi/patches/211-no-kwbimage.patch:7]: Please add a comment to the patch describing why you do not compile `kwbimage.o`.

File: package/boot/uboot-sunxi/patches/300-force-pylibfdt-build.patch [hunk 5]
```diff
@@ -0,0 +1,30 @@
+--- a/Makefile
++++ b/Makefile
+@@ -2000,26 +2000,7 @@ endif
+ # Check dtc and pylibfdt, if DTC is provided, else build them
+ PHONY += scripts_dtc
+ scripts_dtc: scripts_basic
+-	$(Q)if test "$(DTC)" = "$(DTC_INTREE)"; then \
```
Comment (hauke) [inline package/boot/uboot-sunxi/patches/300-force-pylibfdt-build.patch:7]: Please add a comment to the patch why this is needed.

File: package/boot/uboot-sunxi/patches/063-fix-lime2-revK-add-micrel-PHY.patch [hunk 6]
```diff
@@ -11,7 +11,7 @@
  CONFIG_AHCI=y
```
Comment (hauke) [inline package/boot/uboot-sunxi/patches/063-fix-lime2-revK-add-micrel-PHY.patch:11]: This change in `configs/A20-OLinuXino-Lime2-eMMC_defconfig` is already include in upstream U-Boot, see https://github.com/u-boot/u-boot/commit/22885f4ed9f5a67564aa8f0401669a9601051251

The `configs/A20-OLinuXino-Lime2_defconfig` file contains options to activate a Realtek PHY in upstream U-Boot, see https://github.com/u-boot/u-boot/blob/master/configs/A20-OLinuXino-Lime2_defconfig#L23
It only has one Ethernet port, so it is probably using Realtek instead. 

I think you can drop this patch completely.

File: package/boot/uboot-sunxi/patches/003-add-theobroma-a31-pangolin.patch (hunk 1 above)
Comment (wigyori) [reply package/boot/uboot-sunxi/patches/003-add-theobroma-a31-pangolin.patch:371]: I'll drop the Pangolin support completely, it's not even in the target's image Makefile currently - plan to re-add later, but let's drop it for now.

File: package/boot/uboot-sunxi/patches/063-fix-lime2-revK-add-micrel-PHY.patch (hunk 6 above)
Comment (wigyori) [reply package/boot/uboot-sunxi/patches/063-fix-lime2-revK-add-micrel-PHY.patch]: Dropped, indeed this was upstreamed.

File: package/boot/uboot-sunxi/patches/300-force-pylibfdt-build.patch (hunk 5 above)
Comment (wigyori) [reply package/boot/uboot-sunxi/patches/300-force-pylibfdt-build.patch]: Same as in uboot-mediatek.

**Review by hauke — APPROVED** [review]

**Comment by wigyori** [comment]
Thanks all for the reviews.
